spec/app/ics-028-cross-chain-valset-update/README.md

```diff
@@ -0,0 +1,257 @@
+  - Baby blockchain: Baby blockchain is a blockchain that is being secured by the parent blockchain. In other words, validators that secure and operate the baby blockchain are bonded on the parent blockchain. Any misbehavior of a validator at the baby blockchain is punished at the parent blockchain (i.e., the validator is slashed at the parent blockchain).
+  - Causally consistent shared FIFO queue: This queue is used for communication among two blockchains, as we explain in the rest of the document.
+
+Note that the protocol we present is generalized to multiple baby blockchains.
+In other words, a single parent blockchain might have multiple baby blockchains under its "jurisdiction".
+Hence, the Cross-Chain Validation protocol is concerned with more than two entities (one parent blockchain and potentially multiple baby blockchains).
+
+### Properties
+
+This subsection is devoted to defining properties that the Cross-Chain Validation protocol ensures.
+Recall that the parent blockchain has an ability to demand a change to the validator set of the baby chain.
+Moreover, we want to ensure the stake of validators of the baby blockchain are "frozen" at the parent chain.
+
+We present the interface of the protocol:
+  - Request: <ChangeValidatorSet, babyChain, V> - request made by the parent blockchain to change the validator set of babyChain using the validator set updates V.
+  We assume that each validator set update is unique (e.g., each validator set update V could simply have a unique identifier).
+  - Indication: <Unbonded, babyChain, V> - indication to the parent blockchain that the validator set update V (i.e., its "effect") has unbonded on the baby blockchain.
+
+Hence, we aim to achieve the following properties:
+- *Liveness*: If the parent blockchain demands a validator set change V for the baby chain, then the validator set of the baby blockchain reflects this demand.
+- *Validator set change safety*: Suppose that the validator set of the baby blockchain changes from *V* to *V'*. Then, there exists a sequence *seq = V1, V2, ..., Vn*, where *n > 0*, of change validator set demands issued by the parent blockchain such that *apply(V, seq) = V'*.
+- *Validator set change liveness*: If there exist infinitely many <ChangeValidatorSet, babyChain, V'> issued after <ChangeValidatorSet, babyChain, V>, then <Unbonded, babyChain, V> is eventually triggered.
+
+Let us explain the validator set change liveness property.
+First, note that <Unbonded, babyChain, Vlast> is never triggered if Vlast represent the **final** validator set change demand issued by the parent blockchain (the reason is that the validator set of the baby blockchain will never change after Vlast).
```

Comment:
Unbonded acknowledgement should be sent for `Vlast` even if this is the final validator set change demanded by parent chain correct?

The baby chain validator set after `VLast` should be:

`ApplyChanges(VLast, PrevValidatorSet)`

immediately after the validator set change is received.

Applying the last validator set change may cause some baby chain validators to unbond. Thus, we still need to send  the acknowledgement so that the stake on parent chain can be unlocked for any validators that unbonded on the final validator set change.

This should happen regardless independent of whether any future validator set changes come in.

Reply:
I'm just confused by the semantics intended here. If we want the baby chain to control when unbonding happens - so that we can ensure that the baby chain reduces voting power (safely) before a validator is unbonded - can't we just have it acknowledge the validator set updates in some standard fashion (even using IBC-provided acknowledgements, ideally)? Why do we need these separate semantics and infinite messages?